### PR TITLE
Domain to mesh

### PR DESCRIPTION
**Pull request by bmxam**
A bit ugly, but seems to work!

### CODE REVIEW

File: test/mesh/test_domain.jl [hunk 1]
```diff
@@ -20,4 +20,19 @@
...
+        f(x) = norm(x) > 0.5
+        Ω = CellDomain(mesh, f)
+
+        new_mesh = Bcube.domain_to_mesh(Ω)
+
+        @test nnodes(new_mesh) == 8
+        @test ncells(new_mesh) == 3
+        @test Bcube.inner_faces(new_mesh) == [3, 6]
+        @test Bcube.outer_faces(new_mesh) == [1, 2, 4, 5, 7, 8, 9, 10]
+        @test Bcube.boundary_faces(new_mesh, "CLIPPED_BND") == [1, 4, 5]
```
Comment (ghislainb) [inline test/mesh/test_domain.jl:36]: may be one test for original boundary face tags would be great to have

File: src/mesh/domain.jl [hunk 2]
```diff
@@ -16,6 +16,10 @@ abstract type AbstractCellDomain{M, I} <: AbstractDomain{M} end
 A `CellDomain` is a representation of the cells of a mesh. It's primary
 purpose is to represent a domain to integrate over.
 
+# Constructors
+CellDomain(mesh::Mesh)
```
[the remark this answers is not in the document]
Comment (bmxam) [reply src/mesh/domain.jl:20]: done

File: src/mesh/domain.jl [hunk 3]
```diff
@@ -30,8 +34,18 @@ struct CellDomain{M, I} <: AbstractCellDomain{M, I}
     CellDomain(mesh, indices) = new{typeof(mesh), typeof(indices)}(mesh, indices)
 end
 @inline topodim(d::CellDomain) = topodim(get_mesh(d))
+
 CellDomain(mesh::AbstractMesh) = CellDomain(parent(mesh))
 CellDomain(mesh::Mesh) = CellDomain(mesh, 1:ncells(mesh))
+
+function CellDomain(mesh::Mesh, f::Function, all_nodes = true)
```
[the remark this answers is not in the document]
Comment (bmxam) [reply src/mesh/domain.jl]: removed

File: src/mesh/domain.jl [hunk 4]
```diff
@@ -16,6 +16,10 @@ abstract type AbstractCellDomain{M, I} <: AbstractDomain{M} end
 A `CellDomain` is a representation of the cells of a mesh. It's primary
 purpose is to represent a domain to integrate over.
 
+# Constructors
+CellDomain(mesh::Mesh)
+CellDomain(mesh::Mesh, f::Function, all_nodes = true)
```
[the remark this answers is not in the document]
Comment (bmxam) [reply src/mesh/domain.jl]: I've named it `identify_cells`, is it ok for you?

File: src/mesh/domain.jl [hunk 5]
```diff
@@ -555,3 +569,79 @@ function _get_index(domain::BoundaryFaceDomain{M, <:PeriodicBCType}, i::Integer)
 
     return FaceInfo(cellinfo1, cellinfo2, cside_i, cside_j, ftype, fnodes, _f2n, iface)
 end
+
+"""
+    domain_to_mesh(::AbstractDomain)
+
+Convert an `AbstractDomain` to a `Mesh`.
```
[the remark this answers is not in the document]
Comment (bmxam) [reply src/mesh/domain.jl:587]: done + I've added an argument (with default value = CLIPPED_BND) to control the value

File: test/mesh/test_domain.jl (hunk 1 above)
Comment (bmxam) [reply test/mesh/test_domain.jl:37]: please check the new version if it corresponds to what you asked for

**Review by ghislainb — DISMISSED** [review]

**Pull request merged**